code/game/mecha/working/clarke.dm

```diff
@@ -48,6 +48,24 @@
 		var/mob/living/brain/B = M.brainmob
 		hud.add_hud_to(B)
 
+/obj/mecha/working/clarke/play_stepsound(direction)
+	if(prob(30))
+		floor_damage()
```

Comment:
Should this be handled in `play_stepsound`?
That makes for a pretty misleading proc name.

Reply:
The reason I placed it here is because the prior proc in the chain that handles movement is the one that actually checks for step delay. As such, if you hold down the movement key, it would constantly re-roll the chance to break floors even while the mech is on hold in-between steps.

I can look into a better way to do it, though.

Reply:
You could register to `COMSIG_MOVABLE_MOVED` and run your break tile chance proc.

Reply:
Mecha can be moved on MULE? Or in zero gravity. If you use `COMSIG_MOVABLE_MOVED` you need check that you actually move by your treads.

Reply:
I didn't want to use the moved comsig, due to the reasons Dennok mentioned. Moved the code to do_move(), instead.